pkg/operator/ceph/object/controller.go

```diff
@@ -203,12 +213,24 @@ func (r *ReconcileCephObjectStore) reconcile(request reconcile.Request) (reconci
 	// DELETE: the CR was deleted
 	if !cephObjectStore.GetDeletionTimestamp().IsZero() {
 		logger.Debugf("deleting store %q", cephObjectStore.Name)
+		cephObjectStore.Spec.PreservePoolsOnDelete = false
 		cfg := clusterConfig{context: r.context, store: cephObjectStore}
 		err := cfg.deleteStore()
 		if err != nil {
 			return reconcile.Result{}, errors.Wrapf(err, "failed to delete store %q. ", cephObjectStore.Name)
 		}
 
+		// TODO: Add cleanupPolicy for confirming deletion of object buckets.
+		// TODO: Always set PreservePoolsOnDelete to false if object buckets cleanupPolicy is present.
+		if !cephObjectStore.Spec.PreservePoolsOnDelete {
```

Comment:
The PreservePoolsOnDelete setting seems sufficient for whether to delete the pools. I would say we don't need either of these TODOs.
- The cleanupPolicy is for the cluster-wide settings for mons and osds when you're destroying the entire cluster. Here we aren't destroying an entire cluster, we are just deleting an object store.
- If the cleanupPolicy in the cluster CR is set to destroy the cluster, it doesn't seem like we need to worry about that. The pool finalizer can still just worry about its own settings.

Reply:
If I understood the requirement correctly, we need a clean uninstall. When the cluster is deleted (with or without the cleanupPolicy) the OBCs and OBs are still around and we wanted to delete these orphaned resources as well.